### PR TITLE
fix: parsing of tmt context input

### PR DESCRIPTION
**Pull request by jamacku**
Fixes:
- #136

### CODE REVIEW

**Comment by zmiklank** [comment]
[test]

File: src/schema/input.ts [hunk 1]
```diff
@@ -50,9 +50,9 @@ export const tmtArtifactsSchema = z
 // https://testing-farm.gitlab.io/api/#operation/requestsPost
 export const tmtContextSchema = z
   .object({
-    distro: z.string().min(1),
-    arch: z.string().min(1),
-    trigger: z.string().min(1),
+    distro: z.string().min(1).optional(),
```
Comment (zmiklank) [inline src/schema/input.ts:53]: Don't we want to support all tmt context dimensions?
https://tmt.readthedocs.io/en/stable/spec/context.html#dimension

**Review by zmiklank — APPROVED** [review]

**Comment by zmiklank** [comment]
@henrywang We want to have a few days (a week) between landing new fix or feature and release. Meanwhile we test it on our repos.
New tfaga could be released sometime next week then.

**Comment by zmiklank** [comment]
@henrywang Hello, this fix has just landed in v2.

**Comment by henrywang** [comment]
@zmiklank Thank you! I'll give new v2 a try.

**Comment by henrywang** [comment]
@zmiklank Hi! The context issue fixed. But I found the secret can't be passed to tmt.

**Comment by zmiklank** [comment]
@henrywang thanks for confirmation. Do you mean [these secrets](https://github.com/sclorg/testing-farm-as-github-action/blob/cd1f435c152615716e881ad270d19ff06729d371/action.yml#L46)?

**Comment by henrywang** [comment]
> @henrywang thanks for confirmation. Do you mean [these secrets](https://github.com/sclorg/testing-farm-as-github-action/blob/cd1f435c152615716e881ad270d19ff06729d371/action.yml#L46)?

Right.

**Comment by zmiklank** [comment]
Interesting. I can see in pipeline.log in your test run, that secrets were passed in the request:
http://artifacts.osci.redhat.com/testing-farm/f8f506d6-a23f-4f95-9d40-a0e88f2fe6f3/pipeline.log
Also we have an integration test for checking that the secrets input works, and they are still passing:
https://artifacts.dev.testing-farm.io/c8abe438-eba4-4052-84b1-fea4fa5f4e8c/

Can you maybe prove more information about this issue?

**Comment by henrywang** [comment]
The error log:
```
base64: invalid input
base64: invalid input
mv: cannot move '/tmp/tmp.66DLb9AKEw/client.conf' to '/etc/beaker/client.conf': No such file or directory
base64: invalid input
```
Those logs come from https://github.com/virt-s1/bootc-workflow-test/blob/main/tmt/tests/test.sh#L8.

**Comment by zmiklank** [comment]
It seems that decoding base64 ends with success even if its input is zero length string.  Seems more like the binary string you want to pass to the base64 is invalid, not that the variable is empty. If I understand your script correctly, then the base64 command would not be executed at all, if the "secrets" variables were empty.

What is needed here is a minimal reproducer - something that does not incorporate various other programs that could fail. We have something like that here: https://github.com/sclorg/testing-farm-as-github-action/blob/main/.github/workflows/secrets_test.yml, but your issue is not reproducible with it.

**Comment by henrywang** [comment]
@zmiklank I filed a new issue https://github.com/sclorg/testing-farm-as-github-action/issues/152 to track this issue.

**Pull request merged**